pycode/memilio-surrogatemodel/memilio/surrogatemodel/GNN/data_generation_nodamp.py

```diff
@@ -0,0 +1,328 @@
+   """
+    
+    with open(path) as f:
+        data = json.load(f)
+    population = []
+    for data_entry in data:
+        population_county = []
+        population_county.append(
+            data_entry['<3 years'] + data_entry['3-5 years'] / 2)
+        population_county.append(data_entry['6-14 years'])
+        population_county.append(
+            data_entry['15-17 years'] + data_entry['18-24 years'] +
+            data_entry['25-29 years'] + data_entry['30-39 years'] / 2)
+        population_county.append(
+            data_entry['30-39 years'] / 2 + data_entry['40-49 years'] +
+            data_entry['50-64 years'] * 2 / 3)
+        population_county.append(
+            data_entry['65-74 years'] + data_entry['>74 years'] * 0.2 +
+            data_entry['50-64 years'] * 1 / 3)
+        population_county.append(
+            data_entry['>74 years'] * 0.8)
+
+        population.append(population_county)
+    return population
+
+def getBaselineMatrix():
+    """! loads the baselinematrix
+    """
+
+    baseline_contact_matrix0 = os.path.join(
+        "./data/contacts/baseline_home.txt")
+    baseline_contact_matrix1 = os.path.join(
+        "./data/contacts/baseline_school_pf_eig.txt")
+    baseline_contact_matrix2 = os.path.join(
+        "./data/contacts/baseline_work.txt")
+    baseline_contact_matrix3 = os.path.join(
+        "./data/contacts/baseline_other.txt")
+
+    baseline = np.loadtxt(baseline_contact_matrix0) \
+        + np.loadtxt(baseline_contact_matrix1) + \
+        np.loadtxt(baseline_contact_matrix2) + \
+        np.loadtxt(baseline_contact_matrix3)
+
+    return baseline
+
+def generate_data(
+        num_runs, path, input_width, days, save_data=True):
+    """! Generate dataset by calling run_secir_simulation (num_runs)-often
+   @param num_runs Number of times, the function run_secir_simulation is called.
+   @param path Path, where the datasets are stored.
+   @param input_width number of time steps used for model input.
+   @param label_width number of time steps (days) used as model output/label.  
+   @param save_data Option to deactivate the save of the dataset. Per default true.
+   """
+    
+    population = get_population()
+    days_sum = days + input_width - 1
+
+    data = {"inputs": [],
+                "labels": [],
+                }
+
+    # show progess in terminal for longer runs
+    # Due to the random structure, theres currently no need to shuffle the data
+    bar = Bar('Number of Runs done', max=num_runs)
+
+    for _ in range(num_runs):
+
+            data_run = run_secir_groups_simulation(
+                days_sum, population)
+
+            inputs = np.asarray(data_run).transpose(1, 2, 0)[: input_width]
+            data["inputs"].append(inputs)
+
+            data["labels"].append(np.asarray(data_run).transpose(1, 2, 0)[input_width:])
+
+            bar.next()
+
+    bar.finish()
+
+    if save_data:
+            num_groups = int(np.asarray(data['inputs']).shape[2] / 8)
```

Comment:
Check your format on save settings again. The files are indented one position too far and generally a lot of formatting is displayed. :)